contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/gst_transform_resize.py

```diff
@@ -0,0 +1,370 @@
+#!/usr/bin/env python3
+# This file is part of Checkbox.
+#
+# Copyright 2024 Canonical Ltd.
+# Written by:
+#   Patrick Chang <patrick.chang@canonical.com>
+#
+# Checkbox is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# Checkbox is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Checkbox. If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import logging
+import os
+import re
+import shlex
+import subprocess
+
+from abc import ABC, abstractmethod
+from typing import Dict
+
+from checkbox_support.scripts.psnr import get_average_psnr
+
+logging.basicConfig(level=logging.INFO)
+GST_LAUNCH_BIN = os.getenv("GST_LAUNCH_BIN", "gst-launch-1.0")
+OUTPUT_FOLDER = os.getenv("PLAINBOX_SESSION_SHARE", "/var/tmp")
+PLAINBOX_SESSION_SHARE = os.getenv("PLAINBOX_SESSION_SHARE", "/var/tmp")
+VIDEO_CODEC_TESTING_DATA = os.getenv("VIDEO_CODEC_TESTING_DATA")
+if not VIDEO_CODEC_TESTING_DATA or not os.path.exists(
+    VIDEO_CODEC_TESTING_DATA
+):
+    raise SystemExit(
+        "Error: Please define the proper path of golden sample folder to "
+        "the environment variable 'VIDEO_CODEC_TESTING_DATA'"
+    )
+# Folder stores the golden samples
+SAMPLE_2_FOLDER = "sample_2_big_bug_bunny"
+
+
+def register_arguments():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=(
+            "Script helps verify the gst_encoder_psnr scenario of specific"
+            " encoder."
+        ),
+    )
+
+    parser.add_argument(
+        "-p",
+        "--platform",
+        required=True,
+        type=str,
+        help="Json file name which is also the platform e.g. genio-1200",
+    )
+
+    parser.add_argument(
+        "-ep",
+        "--encoder_plugin",
+        required=True,
+        type=str,
```

Comment:
(optional) Looks like the only valid values are `v4l2h264enc` and `v4l2h265enc` so we can specify `options=["v4l2h264enc", "v4l2h265enc"]` and later on when accessing `_codec_parser_map` we don't need to use `.get()`